compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -482,9 +482,11 @@ private void defineFunction(DataInputStream dataInStream) throws IOException {
                             new BAttachedFunction(names.fromString(funcName), invokableSymbol, funcType,
                                     symTable.builtinPos);
                     BStructureTypeSymbol structureTypeSymbol = (BStructureTypeSymbol) attachedType.tsymbol;
-                    if (Names.USER_DEFINED_INIT_SUFFIX.value.equals(funcName)
-                            || funcName.equals(Names.INIT_FUNCTION_SUFFIX.value)) {
-                        structureTypeSymbol.initializerFunc = attachedFunc;
+                    if (Names.USER_DEFINED_INIT_SUFFIX.value.equals(funcName) ||
+                            funcName.equals(Names.INIT_FUNCTION_SUFFIX.value)) {
+                        if (structureTypeSymbol.getKind() == SymbolKind.OBJECT) {
```

Comment:
Why do we need to check this? Won't L485-L486 be true only for objects now?

Reply:
Yes. We can remove it. Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/37184/commits/f846176e6ff38c3e1366f21da4cfa05ea8a4fbdc